lib/system.nim

```diff
@@ -365,7 +365,13 @@ proc arrGet[I: Ordinal;T](a: T; i: I): T {.
 proc arrPut[I: Ordinal;T,S](a: T; i: I;
   x: S) {.noSideEffect, magic: "ArrPut".}
 
-when defined(nimAllowNonVarDestructor):
+when defined(nimAllowNonVarDestructor) and arcLikeMem:
+  proc `=destroy`*(x: string) {.inline, magic: "Destroy".} =
+    discard
```

Comment:
As an side-effort, the string destrcutor gets inlined under ARC/ORC becuase of the pre-existing semOverride implementation.